totalRP3/Modules/Toolbar/Toolbar.lua

```diff
@@ -47,16 +47,39 @@ local function onStart()
 
 	local LDB = LibStub:GetLibrary("LibDataBroker-1.1");
 
+	-- in order to turn a toolbar object into an LDBObject that can be used by other addons (ElvUI) to display information:
+	-- 	modify the object's 'text' attribute to the desired display text
+	-- 	call TRP3_API.toolbar.SignalLDBObjectUpdate, passing the object as the only arg
+	-- 	ensure the default 'text' attribute is a suitable placeholder, as it will be shown temporarily when the object is updated (at least in the case of ElvUI)
+
+	local OBJECT_NAME_FORMAT = TRP3_API.globals.addon_name_short .. " — %s"; -- what is this dash character??
+
+	local function SignalLDBObjectUpdate(object, attr)
+		local name = OBJECT_NAME_FORMAT:format(object.configText);
+
+		-- have to fire both of these events because some addons only listen to one or the other - peak design
+		local callbackNames = {
+			"LibDataBroker_AttributeChanged_" .. name,
+			"LibDataBroker_AttributeChanged_" .. name .. "_" .. attr,
+		};
+		for _, callback in ipairs(callbackNames) do
+			LDB.callbacks:Fire(callback, name, attr, object[attr], object);
+		end
+	end
+	TRP3_API.toolbar.SignalLDBObjectUpdate = SignalLDBObjectUpdate;
```

Comment:
You would probably think that manually firing these callbacks is unnecessary, as the objects seem to emit their own events when their respective fields are updated.

However, for whatever reason, only icon field changes get picked up by LDB displayer addons. Naturally, I tried to fix this by manually triggering the `LibDataBroker_AttributeChanged_<name here>` callback and specifying "text" as the updated field.

Now we've run into a bit of a conundrum. Some addons don't listen to that event, and instead only register callbacks for the slightly more specific version `LibDataBroker_AttributeChanged_<name>_<field>` - resulting in the need to trigger both callbacks for our objects to actually be updated correctly across all* addons.

*only ElvUI and TitanPanel were tested